app/controllers/api/subcollections/snapshots.rb

```diff
@@ -0,0 +1,36 @@
+module Api
+  module Subcollections
+    module Snapshots
+      def snapshots_query_resource(object)
+        object.snapshots
+      end
+
+      def snapshots_create_resource(parent, _type, _id, data)
+        validation = parent.validate_create_snapshot
+
+        if validation[:available]
+          task_id = queue_object_action(
+            parent,
+            "summary",
+            :method_name => "create_snapshot",
+            :args => [data[:name], data[:description], data[:memory]]
+          )
+          action_result(true, "Creating snapshot for #{parent.id}", :task_id => task_id)
+        else
+          action_result(false, validation[:message])
+        end
+      end
+
+      def delete_resource_snapshots(parent, type, id, _data)
+        raise "boom" # no content
+        validation = parent.validate_remove_snapshot(id)
+        if validation[:available]
+          task_id = queue_object_action(parent, "summary", :method_name => "remove_snapshot", :args => [id])
+          action_result(true, "desc", :task_id => task_id)
```

Comment:
"desc" ? maybe "Deleting snapshot #{id} for ..."   same parent signature as above.